src/app/files/fileManager.js

```diff
@@ -101,6 +99,18 @@ class FileManager {
     return self._deps.config.get('currentFile')
   }
 
+  closeFile (name) {
+    var self = this
+    delete self.openedFiles[name]
+    if (Object.keys(self.openedFiles).length) {
+      self.switchFile(Object.keys(self.openedFiles)[0])
+    } else {
+      self._deps.editor.displayEmptyReadOnlySession()
+      self._deps.config.set('currentFile', '')
+    }
+    this.event.trigger('fileClosed', [name])
```

Comment:
Should you use `this.nodeEvent.emit` instead ?

Reply:
yeah, fileManager is already using the remix `EventManager` for the `currentFileChanged` event. better to keep using that one for now and not mixing event...